### ISSUE TITLE
Question: Is ace comprehensive with its suggestions?

### ISSUE THREAD

**Issue by jenstroeger** [daisy/ace#179]
(Ace 1.0.0, Node v6.14.1)

When I check an EPUB with ace, does it check the existing accessibility features _only_ or does ace suggest a (comprehensive) list of missing annotations?

I noticed that ace pointed out missing `role` attributes for existing `epub:type` ones, but I’m curious to what extent and how thoroughly ace checks for missing attributes. For example, when I removed the `epub:type` from a footnote’s `<aside>` then ace wouldn’t issue a warning for that footnote. (Hmm… how would it know that the `<aside>` is supposed to be footnote anyway in plain HTML — I suppose it can’t 🤔)

Or to rephrase my question: if I run an EPUB through ace and ace doesn’t find issues, then that does _not_ mean that the EPUB meets any of the accessibility requirements, correct? This question is perhaps related to the “conformsTo” discussion in issue https://github.com/daisy/ace/issues/171.

**Comment by rdeltour** [comment]
hi @jenstroeger (sorry for the late answer!)

> When I check an EPUB with ace, does it check the existing accessibility features only or does ace suggest a (comprehensive) list of missing annotations?

I’m not sure I understand what you mean by "list of missing annotations"…
Ace intends to only report existing and confirmed accessibility issues.

> I noticed that ace pointed out missing `role` attributes for existing `epub:type` ones, but I’m curious to what extent and how thoroughly ace checks for missing attributes. For example, when I removed the `epub:type` from a footnote’s `<aside>` then ace wouldn’t issue a warning for that footnote. (Hmm… how would it know that the `<aside>` is supposed to be footnote anyway in plain HTML — I suppose it can’t 🤔)

No, without an `epub:type`, Ace would not report any missing role, for the reason you note: it can’t tell whether the `aside` would represent a footnote or something else.

> Or to rephrase my question: if I run an EPUB through ace and ace doesn’t find issues, then that does not mean that the EPUB meets any of the accessibility requirements, correct?

Absolutely correct: Ace will only report the issues that it can detect automatically. An EPUB passing Ace doesn’t necessary meet all the accessibility requirements; many of them require manual inspection!

Hope this helps, and again sorry for having left this issue unnoticed!
Btw, for such questions it’s usually preferable to use the support email address (ace@daisy.org), since this issue tracker is more intended for bugs or feature requests 😉.
